tests/version_test.ts

```diff
@@ -0,0 +1,26 @@
+import { assertEquals } from "../deps.ts";
+
+Deno.test({
+  name: "Version",
+  async fn(): Promise<void> {
+    const p = await Deno.run({
+      cmd: ["deno", "run", "--allow-net", "../../mod.ts", "--version"],
+      cwd: "./tests/up-to-date-deps",
+      stdout: "piped",
+      stderr: "piped",
+    });
+    const status = await p.status();
+    const output = await p.output();
+    await p.close();
+    const stdout = new TextDecoder("utf-8").decode(output);
+    const error = await p.stderrOutput();
+    const stderr = new TextDecoder("utf-8").decode(error);
+    assertEquals(stderr, "");
+    assertEquals(
+      stdout,
+      "dmm 1.0.5\n",
+    );
```

Comment:
I'm skeptical about hardcoded version assertions... That's something we even removed from [Drash](https://github.com/drashland/deno-drash) because it's mehhhh...

I might have a suggestion for this (haven't talked to @ebebbington or @crookse about it). Instead of asserting a specific version, we assert:
1. Output is not empty.
2. Output follows a regex that matches a valid semver string. Link to official regex [here](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string).

Honestly, if we assert number 2, we don't have to assert 1 since that case is covered in 2.

Reply:
@Guergeiro that's a good idea actually

I think the following regex would suffice: `/(v)?[0-9].+[0-9].+[0-9]/g;`

Reply:
I don't see a problem in using the official regex though. They cover alpha releases, release candidates, etc. We literally only have to copy 😆

Reply:
@Guergeiro The regex approach sound seems good to me, I'll add the official regex since it covers all possible cases.

Reply:
Let’s make sure we add a comment stating where we got the regex. I like to know where we get code from when it’s copied.

Reply:
[](https://regexr.com/39s32)

What do you guys think about this? I guess this should do the work.

Reply:
I would prefer if it was done separately. Pseudocode:
```typescript
const reg = /^((([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$/;
const variable = "dmm 1.5.5";
const splittedString = variable.split(" ");
assertTrue(splittedString.length == 2);
assertEquals(splittedString[0], "dmm");
assertTrue(reg.test(splittedString[1]);
```